src/extension.c

```diff
@@ -201,6 +202,10 @@ extension_is_loaded(void)
 		if (OidIsValid(extension_oid) && extension_oid == CurrentExtensionObject)
 			return false;
 	}
+	if (guc_restoring)
```

Comment:
Is this better placed before `if (creating_extension)`? If `guc_restoring` is true, it shouldn't be necessary to execute that if clause.